File: result/translate/transcoder/java2cpp/c++/LEXICOGRAPHICALLY_MINIMUM_STRING_ROTATION.cpp
string f_gold ( string str ) {
  int n = str . size ( ) ;
  string arr [ n ] ;
  string concat = str + str ;
  for ( int i = 0 ;
  i < n ;
  i ++ ) arr [ i ] = concat . substr ( i , n - i ) ;
  sort ( arr , arr + n ) ;
  return arr [ 0 ] ;
}

File: result/translate/starchat/python2java/java/LEXICOGRAPHICALLY_MINIMUM_STRING_ROTATION.java
public static String f_gold ( String str ) {
        int n = str. length ( );
        String [ ] arr = new String [ n ];
        String concat = str + str;
        for ( int i = 0; i < n; i++ ) {
            arr [ i ] = concat.substring ( i, n + i );
        }
        Arrays. sort ( arr );
        return arr [ 0 ];
    }